sympy/integrals/manualintegrate.py

```diff
@@ -1910,7 +1911,7 @@ def substitution_rule(integral):
                         could_be_zero.append(denom)
 
                     for expr in could_be_zero:
-                        if not fuzzy_not(expr.is_zero):
+                        if not fuzzy_not(expr.is_zero) and not integrand.has_free(Pow(expr, S.NegativeOne)):
```

Comment:
How do we know it is sufficient to check specfically for `1/expr` rather than say `1/expr**2` or something?

There s a function `denoms` somewhere that is used for this sort of purpose in `solve`.